src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -483,61 +527,61 @@ private class SamRecordWriter(prefix: PathPrefix,
 }
 
 private[fastq] object FastqRecordWriter {
-  private[fastq] def extensions(pairedEnd: Boolean, illuminaStandards: Boolean = false): Seq[String] = (pairedEnd, illuminaStandards) match {
-    case (true, true)   => Seq("_R1_001.fastq.gz", "_R2_001.fastq.gz")
-    case (true, false)  => Seq("_R1.fastq.gz", "_R2.fastq.gz")
-    case (false, true)  => Seq("_R1_001.fastq.gz")
-    case (false, false) => Seq(".fastq.gz")
+  private[fastq] def extensions(pairedEnd: Boolean, illuminaFileNames: Boolean): Seq[String] = {
+    (pairedEnd, illuminaFileNames) match {
+      case (true, true)   => Seq("_R1_001.fastq.gz", "_R2_001.fastq.gz")
+      case (true, false)  => Seq("_R1.fastq.gz", "_R2.fastq.gz")
+      case (false, true)  => Seq("_R1_001.fastq.gz")
+      case (false, false) => Seq(".fastq.gz")
+    }
   }
 }
 
 /** A writer that writes [[DemuxRecord]]s as [[FastqRecord]]s. */
-private[fastq] class FastqRecordWriter(prefix: PathPrefix, val pairedEnd: Boolean, val illuminaStandards: Boolean = false) extends DemuxWriter {
-  private val writers: IndexedSeq[FastqWriter] = FastqRecordWriter.extensions(pairedEnd=pairedEnd, illuminaStandards=illuminaStandards).map { ext =>
-    FastqWriter(Io.toWriter(PathUtil.pathTo(s"${prefix}${ext}")))
-  }.toIndexedSeq
-
-  private[fastq] def readName(rec: DemuxRecord): String = {
-    if (illuminaStandards) {
-      val comment = rec.comment.getOrElse {
-        throw new IllegalArgumentException(s"Comment required with illumina-standards and read: '${rec.name}'")
-      }
-
-      require(rec.name.count(_ == ':') == 6,
-        s"Expected the read name format 'Instrument:RunID:FlowCellID:Lane:Tile:X:Y', found '${rec.name}'")
-      require(comment.count(_ == ':') == 3,
-        s"Expected the comment format 'ReadNum:FilterFlag:0:SampleNumber', found '${rec.name}'")
+private[fastq] class FastqRecordWriter(prefix: PathPrefix, val pairedEnd: Boolean, val fastqStandards: FastqStandards) extends DemuxWriter {
+  private val writers: IndexedSeq[FastqWriter] = {
+    FastqRecordWriter.extensions(pairedEnd=pairedEnd, illuminaFileNames=fastqStandards.illuminaFileNames).map { ext =>
+      FastqWriter(Io.toWriter(PathUtil.pathTo(s"$prefix$ext")))
+    }.toIndexedSeq
+  }
 
-      s"${rec.name} $comment"
+  def add(rec: DemuxRecord): FastqRecord = {
+    val name = {
+      if (fastqStandards.includeSampleBarcodes) rec.name // the comment is set below, so don't include it here
+      else Seq(Some(rec.name), rec.sampleBarcode, rec.molecularBarcode).flatten.mkString(":")
     }
-    else {
-      Seq(Some(rec.name), rec.sampleBarcode, rec.molecularBarcode).flatten.mkString(":")
+    val comment = rec.readInfo match {
+      case None       => rec.comment
+      case Some(info) =>
+        if (fastqStandards.includeSampleBarcodes && rec.sampleBarcode.nonEmpty) {
+          Some(info.copy(sampleInfo=rec.sampleBarcode.mkString("+")).toString)
+        }
+        else Some(info.toString)
```

Comment:
I'm not really sure I'm following what's going on here.  A couple of comments to clarify intent would be helpful.